src/qt/walletmodel.cpp

```diff
@@ -668,3 +686,210 @@ bool WalletModel::saveReceiveRequest(const std::string &sAddress, const int64_t
     else
         return wallet->AddDestData(dest, key, sRequest);
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+
+    try {
+        res = name_show( params, false);
```

Comment:
In general, I like the way you try to avoid code duplication using the RPC interface.  In this case, however, I think that just querying the name database directly is not too complicated.  You can try to do that instead of calling `name_show` - unless you prefer to call `name_show` nevertheless and think this is easier for you.  I'll leave this decision to you.